onnxruntime/core/providers/rocm/math/softmax_tunable_op.cuh

```diff
@@ -0,0 +1,79 @@
+                                                                   params->softmax_elements, params->input_stride,
+                                                                   params->output_stride);
+  } else {
+    softmax_block_forward<VecSize, input_t, acc_t, output_t, SoftMaxForwardEpilogue>
+        <<<grid, block, block.x * sizeof(acc_t), params->stream>>>(params->output, const_cast<input_t*>(params->input),
+                                                                   params->softmax_elements, params->input_stride,
+                                                                   params->output_stride);
+  }
+  return HIP_CALL(hipGetLastError());
+}
+
+template <typename input_t, typename output_t, typename acc_t>
+Status SoftmaxBlockwiseStaticSelection(const SoftmaxParams<input_t, output_t>* params) {
+  dim3 grid(params->batch_count);
+  constexpr int ILP = sizeof(float4) / sizeof(input_t);
+  dim3 block = SoftMax_getBlockSize(ILP, params->softmax_elements);
+  if (params->is_log_softmax) {
+    softmax_block_forward<ILP, input_t, acc_t, output_t, LogSoftMaxForwardEpilogue>
+        <<<grid, block, block.x * sizeof(acc_t), params->stream>>>(params->output, const_cast<input_t*>(params->input),
+                                                                   params->softmax_elements, params->input_stride,
+                                                                   params->output_stride);
+  } else {
+    softmax_block_forward<ILP, input_t, acc_t, output_t, SoftMaxForwardEpilogue>
+        <<<grid, block, block.x * sizeof(acc_t), params->stream>>>(params->output, const_cast<input_t*>(params->input),
+                                                                   params->softmax_elements, params->input_stride,
+                                                                   params->output_stride);
+  }
+  return HIP_CALL(hipGetLastError());
+}
+
+template <typename input_t, typename output_t, typename acc_t>
+class SoftmaxTunableOp : public onnxruntime::rocm::tunable::TunableOp<SoftmaxParams<input_t, output_t>> {
+ public:
+  SoftmaxTunableOp() {
+    this->RegisterOp(SoftmaxBlockwiseStaticSelection<input_t, output_t, acc_t>);
+    this->RegisterOp(SoftmaxBlockwiseOp<input_t, output_t, acc_t, 1>);
+    this->RegisterOp(SoftmaxBlockwiseOp<input_t, output_t, acc_t, 2>);
+    this->RegisterOp(SoftmaxBlockwiseOp<input_t, output_t, acc_t, 4>);
+    this->RegisterOp(SoftmaxBlockwiseOp<input_t, output_t, acc_t, 8>);
+    this->RegisterOp(SoftmaxBlockwiseOp<input_t, output_t, acc_t, 16>);
+
+#ifdef USE_COMPOSABLE_KERNEL
+    for (auto&& [_, op] : GetCKSoftmaxTypeStringAndOps<input_t, output_t, acc_t>()) {
+      ORT_UNUSED_PARAMETER(_);
+      this->RegisterOp(std::move(op));
+    }
+#endif  // USE_COMPOSABLE_KERNEL
+
+    // NOTE: the 1st kernel is SoftmaxBlockwise Original implementation.
+    this->SetDefaultId(0);
```

Comment:
> `this->SetDefaultId(0);`

Is this necessary? If left unspecified, I believe that `0` is automagically the default implementation that is chose.

Reply:
Yes, it's not necessary here.